addons/mail/models/mail_thread.py

```diff
@@ -4592,6 +4592,9 @@ def _message_update_content(self, message, body, attachment_ids=None, partner_id
         if msg_values:
             message.write(msg_values)
 
+        if message._filter_empty() and message.link_preview_ids:
+            message.link_preview_ids._unlink_and_notify()
```

Comment:
You can just do an `unlink` here. And add the `link_preview_ids` in the `res` so that all the updated data is sent in one go.